Experimental/src/org/sleuthkit/autopsy/experimental/textclassifier/TextClassifierUtils.java

```diff
@@ -0,0 +1,212 @@
+            return new String[0];
+        }
+
+        //Read the file and tokenize it.
+        try {
+            //Read the text
+            String text = IOUtils.toString(reader);
+            return cleanAndTokenize(text);
+        } catch (IOException ex) {
+            LOGGER.log(Level.WARNING, "Cannot extract tokens from file " + file.getName(), ex);
+            return new String[0];
+        }
+    }
+
+    private static String[] cleanAndTokenize(String text) throws IOException {
+        //Initialize the sentenceDetector if that hasn't been done yet.
+        if (sentenceDetector == null) {
+            //Define the sentence detector, trained on the train section of the EWT corpus, from Universal Dependencies
+            final File sentenceModelFile = InstalledFileLocator.getDefault().locate("textclassifier/en-sent-ewt.bin", TextClassifierUtils.class.getPackage().getName(), false); //NON-NLS
+            if (sentenceModelFile == null) {
+                throw new IOException("Error finding sentence detector module");
+            }
+            sentenceDetector = new NewlineHeuristicSentenceDetector(new SentenceDetectorME(new SentenceModel(new FileInputStream(sentenceModelFile))));
+        }
+        ArrayList<String> tokens = new ArrayList<>();
+        String[] sentences = sentenceDetector.sentDetect(text);
+        for (String sentence : sentences) {
+            //Tokenize the file.
+            String[] sentenceTokens = TOKENIZER.tokenize(sentence);
+            ArrayList<String> tokensToKeep = new ArrayList<>();
+            if (sentenceTokens.length > 5) {
+                tokens.addAll(Arrays.asList(sentenceTokens));
+            }
+        }
+        return tokens.toArray(new String[0]);
+    }
+
+    /**
+     * Loads a Naive Bayes categorizer from disk.
+     *
+     * @return the categorizer
+     * @throws IOException if the model cannot be found on disk, or if the file
+     * does not seem to be a model file
+     */
+    static DocumentCategorizerME loadCategorizer() throws IOException {
+        ensureTextClassifierFolderExists();
+        FileReader fr = new FileReader(new File(MODEL_PATH));
+        NaiveBayesModelReader reader = new PlainTextNaiveBayesModelReader(new BufferedReader(fr));
+        reader.checkModelType();
+        NaiveBayesModel model = (NaiveBayesModel) reader.constructModel();
+        fr.close();
+        DoccatModel doccatModel = new DoccatModel(LANGUAGE_CODE, model, new HashMap<>(), new DoccatFactory());
+        return new DocumentCategorizerME(doccatModel);
+    }
+
+    /**
+     * Writes a naive Bayes classifier model to disk.
+     *
+     * @param model the model
+     * @throws IOException If the model file cannot be written. Large files can
+     * cause this.
+     */
+    static void writeModel(NaiveBayesModel model) throws IOException {
+        ensureTextClassifierFolderExists();
+        FileWriter fw = new FileWriter(new File(MODEL_PATH));
```

Comment:
When this throws an exception the file writer may not get closed, consider a try with resources.